### DISCUSSION TITLE
Calling deepgram API locally

### DISCUSSION THREAD

**Discussion by samuelsurulere** [deepgram/community#748] [General help] [answered]
I am trying to use to deepgram API to access the transcription and speakers of a local audio file. The first code sample I used was the SDK code available for transcribing local files

from deepgram import (
    DeepgramClient,
    PrerecordedOptions,
    FileSource,
)
import csv

AUDIO_FILE = "C:/Users/samue/Downloads/Upzoning Day 2 - Afternoon 2.mp3"

def main():
    try:
        deepgram = DeepgramClient("")
        
        with open(AUDIO_FILE, "rb") as file:
            buffer_data = file.read()
        
        payload: FileSource = {
            "buffer": buffer_data,
        }
        
        options = PrerecordedOptions(
            model="nova-2",
            language="en",
            smart_format=True,
            # diarize=True,
        )
        
        response = deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)
        extracted_transcript = response["results"]["channels"][0]["alternatives"][0]["transcript"]
        with open('public_hearings_deepgram.csv', 'w', newline='') as csvfile:
            writer = csv.writer(csvfile)
            writer.writerow(['Speaker', 'Text'])
            writer.writerow(['Text'])
            writer.writerow([extracted_transcript])
        print("File transcribed successfully and saved to public_hearings_deepgram.csv")
        print(response.to_json(indent=4))
    
    except Exception as e:
        print(f"Exception: {e}")

if __name__ == "__main__":
    main()

This code didn't not let me use the diarize=True option. It always returned an Operation timed out exception. I then tried the code for the requests library

import requests, csv

url = "https://api.deepgram.com/v1/listen?smart_format=true&diarize=true&utterances=true&language=en&model=nova-2"

headers = {
    "Authorization": "Token ",
    "Content-Type": "audio/*"
}

with open("C:/Users/samue/Downloads/Upzoning Day 2 - Afternoon 2.mp3", "rb") as audio_file:
    # Make the HTTP request
    response = requests.post(url, headers=headers, data=audio_file)

if response.status_code == 200:
    with open('transcript.csv', 'w', newline='') as csvfile:
        writer = csv.writer(csvfile)
        
        writer.writerow(['Speaker', 'Text'])
        
        data = response.json()
        
        for utterance in data['results']['utterances']:
            speaker_name = utterance['speaker']
            text = utterance['transcript']
            writer.writerow([speaker_name, text])
    print("Transcription saved to transcript.csv")
else:
    print(f"Error: API request failed with status code {response.status_code}")

print(response.json())

This code block was giving entirely different results and the transcript text was of lower quality compared to the API Playground demo. I have tried adjusting the parameters but still stuck with low quality results. I would really appreciate some help.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by davidvonthenen** [comment, marked as answer]
Hi @samuelsurulere 

If your files are large in size or content, you may have to increase the timeout length on the operation. You can do so like this:
https://developers.deepgram.com/docs/dotnet-sdk-pre-recorded-transcription#increasing-the-timeout-for-processing-larger-files
